lib/hanami/application.rb

```diff
@@ -339,7 +339,8 @@ def initialize(application = self.class)
 
         resolver = application.config.router.resolver.new(
           slices: application.slices,
-          inflector: application.inflector
+          inflector: application.inflector,
+          endpoints_namespace: application.config.router.endpoints_namespace
```

Comment:
@waiting-for-dev wouldn't it make more sense to call this `action_namespace`? (or `actions_namespace` I never know if plural is better in such cases, @timriley?)

Reply:
Yeah, I always have existential thoughts about singular/plural naming :rofl: I went with plural because, in the same class, we have a `slices_registry` setting, so it's more consistent. But we can rename both. In general, I prefer singular because they are shorter.

About `actions_namespace` vs. `endpoints_namespace`, I felt that the latter is less opinionated. I mean, we can say that, by definition, that's an endpoint, as a router routes from something to a destination or point. On the other side, calling that an `action` is more of a convention. By I'm 100% fine with either way :slightly_smiling_face:

Reply:
Regarding naming, one thing to consider would be whether there's anything else inside the framework in general that might want to know about our "actions namespace". Would we want to be able to configure this such that generators etc. respect it?

If that was the case, then I think it might make sense as a top-level `config.actions_namespace` settings.

However, if we don't have a clear case for that kind of usage, then I think nesting it inside `config.router` makes sense, and at that point, so does using the same terminology as the router itself, so that'd be `config.router.endpoints_namespace`.

Perhaps we should start with the latter, and then if we discover the need for something more framework-wide, we can consider moving things at that point.

Regarding singular/plural, I think singular _probably_ feels better? I'd be happy for @waiting-for-dev to do that and update the `slices_registry` to make it consistent :)

Reply:
@timriley I'd already make it a top level setting, so `config.actions_namespace`, because it's not just a router concern. Generators will use this already.

Reply:
@solnic @timriley @jodosha finally I updated the PR:

- Rebased from `main`
- Made it a top level setting
- Singularized from `endpoints_namespace` => `endpoint_namespace`
- Renamed `slices_registry` => `slice_registry` for consistency

I also had to change `Hanami::Configuration.components_dir_path` to be an instance method for now, as it needs access to the new setting. My first attempt was also to add two new settings, `repository_namespace` and `view_namespace`, but I realized it makes no sense at this point. Unless I'm wrong, the view namespace is now controlled by `hanami/action`, and repositories are somehow more optional in the stack. Probably we should try to gain some consistency here :)